tests/e2e/vmlistener_test.go

```diff
@@ -346,6 +352,159 @@ func (s *VMListenerTestParams) TestFailoverAcrossVmOnVsan(c *C) {
 	misc.LogTestEnd(c.TestName())
 }
 
+// TestVolumeAttachedForHostdRestart - verify an in use volume
+// remains attached when hostd is restarted.
+// 1. Restart hostd
+// 2. Verify volume stays as attached
+// 3. Kill the vm
+// 4. Status of the volume should be detached.
+func (s *VMListenerTestParams) TestVolumeAttachedForHostdRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create the volume
+	out, err := dockercli.CreateVolume(s.vm1, s.volumeName)
```

Comment:
@shaominchen can you in that case restore that part of the code the way it was before  #1550? I had to redo this change to handle the way the code was redone in #1550. 

@shuklanirdesh82 was this considered when approving the changes for #1550.

I fully agree its disgusting to have to keep refactoring this code over and over with each developer deciding to make changes in one way and then turning around and commenting on it all over!

Let me know if you'll can revert the change to this file and I'd be glad to make the revert the set up per test.